crates/api_models/src/admin.rs

```diff
@@ -273,6 +300,117 @@ pub struct MerchantDetails {
     /// The merchant's address details
     pub address: Option<AddressDetails>,
 }
+#[cfg(feature = "payouts")]
+pub mod payout_routing_algorithm {
+    use std::{fmt, str::FromStr};
+
+    use serde::{
+        de::{self, Visitor},
+        Deserializer,
+    };
+    use serde_json::Map;
+
+    use super::PayoutRoutingAlgorithm;
+    use crate::enums::PayoutConnectors;
+    struct RoutingAlgorithmVisitor;
+    struct OptionalRoutingAlgorithmVisitor;
+
+    impl<'de> Visitor<'de> for RoutingAlgorithmVisitor {
```

Comment:
We'd preferably want to have unit tests for this `Visitor` implementation. Can be done in a separate PR.

Reply:
Sure